x-pack/plugins/infra/server/lib/snapshot/query_helpers.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { metricAggregationCreators } from './metric_aggregation_creators';
+import { InfraSnapshotRequestOptions } from './snapshot';
+
+export const getGroupedNodesSources = (options: InfraSnapshotRequestOptions) => {
+  const sources = [];
+  options.groupby.forEach(gb => {
```

Comment:
Agreed on the `map`, but I'm not sure about the return type -- this function is used only once and only factored out for readability. Adding an explicit type only adds mental overload here in my opinion.

Reply:
I only recommended the type because Typescript will likely complain about the object/key signature. If it works without the type then it should be fine.